src/tests/conformance/post_handshake/validators.rs

```diff
@@ -55,3 +85,212 @@ async fn c015_TM_VALIDATOR_LIST_COLLECTION_node_should_send_validator_list() {
+    let mut node = Node::builder()
+        .log_to_stdout(true)
+        .start(target.path(), NodeType::Stateless)
+        .await
+        .expect("unable to start stateful node");
+
+    let mut test_config = TestConfig::default();
+    test_config.synth_node_config.generate_new_keys = false;
+    let synth_node = SyntheticNode::new(&test_config).await;
+
+    synth_node
+        .connect(node.addr())
+        .await
+        .expect("unable to connect");
+
+    // 1. Setup keys & prefix.  Both master and signing key pairs have been previously generated.
+    let master_secret_hex = "8484781AE8EEB87D8A5AA38483B5CBBCCE6AD66B4185BB193DDDFAD5C1F4FC06";
+    // The master public key should be in the validators.txt file, in ~/.ziggurat/ripple/setup
+    let master_public_hex = "02ED521B8124454DD5B7769C813BD40E8D36E134DD51ACED873B49E165327F6DF2";
+    let master_secret_bytes = hex::decode(master_secret_hex).expect("unable to decode hex");
+    let master_public_bytes = hex::decode(master_public_hex).expect("unable to decode hex");
+    let master_secret_key =
+        SecretKey::from_slice(master_secret_bytes.as_slice()).expect("unable to create secret key");
+
+    let signing_secret_hex = "00F963180681C0D1D51D1128096B8FF8668AFDC41CBDED511D12D390105EFDDC";
+    let signing_public_hex = "03859B76317C8AA64F2D253D3547831E413F2663AE2568F7A17E85B283CC8861E4";
+    let signing_secret_bytes = hex::decode(signing_secret_hex).expect("unable to decode hex");
+    let signing_public_bytes = hex::decode(signing_public_hex).expect("unable to decode hex");
+    let signing_secret_key = SecretKey::from_slice(signing_secret_bytes.as_slice())
+        .expect("unable to create secret key");
+
+    // 2. Create signable manifest with sequence, public key, signing public key (without signatures)
+    assert_eq!(
+        master_public_bytes.len(),
+        PUBLIC_KEY_SIZE,
+        "invalid master public key length: {}",
+        master_public_bytes.len()
+    );
+    assert_eq!(
+        signing_public_bytes.len(),
+        PUBLIC_KEY_SIZE,
+        "invalid signing public key length: {}",
+        master_public_bytes.len()
+    );
+    let signable_manifest =
+        create_signable_manifest(1, &master_public_bytes, &signing_public_bytes);
+
+    // 3. append manifest prefix
+    let mut prefixed_signable = BytesMut::with_capacity(1024);
+    prefixed_signable.put(&b"MAN\x00"[..]);
+    prefixed_signable.extend_from_slice(&signable_manifest);
+
+    // 4. Sign the signable manifest with master secret key, get master signature
+    let master_signature_bytes = sign_buffer(&master_secret_key, &prefixed_signable);
+
+    // 5. Sign it with signing private key, get signature
+    let signature_bytes = sign_buffer(&signing_secret_key, &prefixed_signable);
+
+    // 6. Create final manifest with sequence, public key, signing public key, master signature, signature
+    let manifest = create_final_manifest(
+        1,
```

Comment:
General improvement 2:

Just like they did in [their code](https://github.com/XRPLF/rippled/blob/master/src/ripple/protocol/impl/Sign.cpp#L25), we could handle the prefix from the `sign_buffer` function, e.g.:
```
let master_signature_bytes = sign_buffer(&hash_prefix, &master_secret_key, &manifest);
// or just call it `sign` instead of `sign_buffer` to be aligned with their code base
```


------
Then the current `create_final_manifest` function could be renamed to `sign_manifest`, and we could reuse the old manifest here (also make the `create_final_manifest` function smaller since we would only need to append `master_signature_bytes` and `signature_bytes` from that function to our already existing `manifest`):
```
sign_manifest(&mut manifest, &master_signature_bytes, &signature_bytes);
```